pkg/frontend/querymiddleware/codec.go

```diff
@@ -334,18 +334,24 @@ func (Codec) MergeResponse(responses ...Response) (Response, error) {
 }
 
 // DecodeMetricsQueryRequest decodes a MetricsQueryRequest from an http request.
-func (c Codec) DecodeMetricsQueryRequest(_ context.Context, r *http.Request) (MetricsQueryRequest, error) {
+func (c Codec) DecodeMetricsQueryRequest(ctx context.Context, r *http.Request) (MetricsQueryRequest, error) {
```

Comment:
Rather than refactoring `Codec`, what if we used `Codec` as it was before this PR from query-tee?